man/assign.Rd

```diff
@@ -88,6 +88,8 @@ Since \code{[.data.table} incurs overhead to check the existence and type of arg
     \code{DT[a > 4, b := c]} is different from \code{DT[a > 4][, b := c]}. The first expression updates (or adds) column \code{b} with the value \code{c} on those rows where \code{a > 4} evaluates to \code{TRUE}. \code{X} is updated \emph{by reference}, therefore no assignment needed.
 
     The second expression on the other hand updates a \emph{new} \code{data.table} that's returned by the subset operation. Since the subsetted data.table is ephemeral (it is not assigned to a symbol), the result would be lost; unless the result is assigned, for example, as follows: \code{ans <- DT[a > 4][, b := c]}.
+
+    Regarding subset operations: \code{data.table} only creates a copy of its subset when the subset operation is non-empty. For example, \code{DT[a > 4]} returns a copy of \code{DT} where \code{a > 4} is \code{TRUE}. However, \code{DT[]} returns a reference to the original \code{DT}.
```

Comment:
looks ok overall but I'm not sure about "subset operation is non-empty" maybe change to "`i` argument is provided" ?